pandas/tseries/offsets.py

```diff
@@ -184,10 +185,17 @@ def __add__(date):
     )
     _use_relativedelta = False
     _adjust_dst = False
+    _typ = "dateoffset"
 
     # default for prior pickles
     normalize = False
 
+    def __setattr__(self, key, value):
+        if key in ['n', '_offset'] and hasattr(self, key):
+            raise TypeError('%s is intended to be immutable; "%s" '
```

Comment:
Eventually.  First there are some lingering corner cases in which the current implementation can be incorrect.